### PR TITLE
Exclude 'no-entity' from NER evaluation

### LINKED ISSUE #4313: Evaluation of NER: Don't include 'no-entity'
Currently, we take the weighted precision and f1 score as performance metric for our named entity recognition. Weighted means

> Calculate metrics for each label, and find their average weighted
> by support (the number of true instances for each label). This
> alters 'macro' to account for label imbalance; it can result in an
> F-score that is not between precision and recall. 

In most cases, ~90% of all words have the label 'no-entity'. If we include 'no-entity' as a label in the evaluation, this will influence the result a lot. The overall metrics are therefore always quite high. If you want to get a feeling on how good your NER is actually doing, you need to look at the metrics for the labels.

Solution:
Report the overall metrics without taking the label 'no-entity' into account. We can still list the precision, recall, f1 score for 'no-entity' separately in the report, but it should not influence the overall metrics anymore.

### PR DESCRIPTION
**Pull request by tabergma**
**Proposed changes**:
- evaluation of NER does not consider 'no-enity' anymore
closes #4313 

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

### CODE REVIEW

File: rasa/nlu/test.py [hunk 1]
```diff
@@ -146,24 +148,38 @@ def log_evaluation_table(
 
 
 def get_evaluation_metrics(
-    targets: Iterable[Any], predictions: Iterable[Any], output_dict: bool = False
+    targets: Iterable[Any],
+    predictions: Iterable[Any],
+    output_dict: bool = False,
+    include_no_entity: bool = True,
```
Comment (Ghostvv) [inline rasa/nlu/test.py]: shall we use the same default everywhere?

File: rasa/nlu/test.py [hunk 2]
```diff
@@ -146,24 +148,38 @@ def log_evaluation_table(
...
     )
-    precision = metrics.precision_score(targets, predictions, average="weighted")
-    f1 = metrics.f1_score(targets, predictions, average="weighted")
+    f1 = metrics.f1_score(targets, predictions, labels=labels, average="weighted")
     accuracy = metrics.accuracy_score(targets, predictions)
 
     return report, precision, f1, accuracy
 
 
+def get_label_set(targets: Iterable[Any], include_no_entity: bool):
```
Comment (Ghostvv) [inline rasa/nlu/test.py]: please add method description

File: rasa/nlu/test.py [hunk 3]
```diff
@@ -411,31 +427,34 @@ def evaluate_entities(
...
+
             logger.info(
                 "Classification report for '{}' saved to '{}'."
                 "".format(extractor, extractor_report_filename)
             )
 
         else:
             report, precision, f1, accuracy = get_evaluation_metrics(
-                merged_targets, merged_predictions
+                merged_targets, merged_predictions, include_no_entity=False
```
Comment (Ghostvv) [inline rasa/nlu/test.py]: should we explicitly say `output_dict=False,`? took me some time to get the difference

File: rasa/nlu/test.py (hunk 1 above)
Comment (tabergma) [reply rasa/nlu/test.py]: The function is also used to evaluate intents. That is why I added the default value.

File: rasa/nlu/test.py (hunk 3 above)
Comment (tabergma) [reply rasa/nlu/test.py]: No need for that, default is `False`.

File: rasa/nlu/test.py (hunk 1 above)
Comment (Ghostvv) [reply rasa/nlu/test.py]: I see, maybe invert and rename `include_no_entity` to smth like `exclude_label: Text`?

File: rasa/nlu/test.py (hunk 3 above)
Comment (Ghostvv) [reply rasa/nlu/test.py]: yes, for clarification

File: rasa/nlu/test.py [hunk 4]
```diff
@@ -173,7 +176,7 @@ def get_evaluation_metrics(
     return report, precision, f1, accuracy
 
 
-def get_label_set(targets: Iterable[Any], exclude_label: Optional[Text]):
+def get_label_set(targets: Iterable[Any], exclude_label: Optional[Text]) -> List[Text]:
```
Comment (Ghostvv) [inline rasa/nlu/test.py]: `get_unique_labels`?

**Pull request merged**